File: tester/src/accounts/outputs.rs
// Copyright 2022 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

use iota_wallet::account::AccountHandle;
use serde_json::Value;

use crate::{accounts::amount::process_amount, context::Context, error::Error};

pub async fn process_outputs<'a>(
    context: &Context<'a>,
    account: &AccountHandle,
    outputs: &[Value],
) -> Result<(), Error> {
    for output in outputs {
        if output.is_u64() {
            process_amount(context, account, output.as_u64().unwrap()).await?;
        }
    }

    Ok(())
}

File: bindings/java/lib/src/main/java/org/iota/types/account_methods/Outputs.java
// Copyright 2023 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package org.iota.types.account_methods;

import org.iota.types.FilterOptions;

/// Returns all outputs of the account
public class Outputs implements AccountMethod {

    /// Options to filter outputs
    private FilterOptions filterOptions;

    public Outputs withFilterOptions(FilterOptions filterOptions) {
        this.filterOptions = filterOptions;
        return this;
    }
}